apps/web/src/components/Basenames/UsernameProfileSectionFrames/Context.tsx

```diff
@@ -0,0 +1,277 @@
+  const currentChainId = useChainId();
+  const config = useConfig();
+  const { openConnectModal } = useConnectModal();
+  const [frameInteractionError, setFrameInteractionError] = useState('');
+  const onTransaction: OnTransactionFunc = useCallback(
+    async ({ transactionData }) => {
+      if (!address) {
+        openConnectModal?.();
+        return null;
+      }
+
+      try {
+        const { params, chainId } = transactionData;
+        const requestedChainId = parseChainId(chainId);
+
+        if (currentChainId !== requestedChainId) {
+          await switchChain(config, {
+            chainId: requestedChainId,
+          }).catch((e) => {
+            throw new CouldNotChangeChainError(e.message as string);
+          });
+        }
+
+        const transactionId = await sendTransaction(config, {
+          to: params.to,
+          data: params.data,
+          value: BigInt(params.value ?? 0),
+        });
+        return transactionId;
+      } catch (error) {
+        if (error instanceof InvalidChainIdError) {
+          setFrameInteractionError('Invalid chain id');
+        } else if (error instanceof CouldNotChangeChainError) {
+          setFrameInteractionError('Could not change chain');
+        } else {
+          setFrameInteractionError('Error sending transaction');
+        }
+
+        console.warn(error);
+
+        return null;
+      }
+    },
+    [address, config, currentChainId, openConnectModal],
+  );
+  const onError = useCallback((e: Error) => {
+    console.error(e);
+  }, []);
+  const onSignature: OnSignatureFunc = useCallback(
+    async ({ signatureData }) => {
+      if (!address) {
+        openConnectModal?.();
+        console.info('Opened connect modal because the account address is not set');
+
+        return null;
+      }
+
+      try {
+        const { params, chainId } = signatureData;
+        const requestedChainId = parseChainId(chainId);
+
+        if (currentChainId !== requestedChainId) {
+          await switchChain(config, {
+            chainId: requestedChainId,
+          }).catch((e) => {
+            throw new CouldNotChangeChainError(e.message as string);
+          });
+        }
+
+        return await signTypedData(config, params);
+      } catch (error) {
+        if (error instanceof InvalidChainIdError) {
+          setFrameInteractionError('Invalid chain id');
+        } else if (error instanceof CouldNotChangeChainError) {
+          setFrameInteractionError('Could not change chain');
+        } else {
+          setFrameInteractionError('Error signing data');
+        }
+
+        console.error(error);
+
+        return null;
+      }
+    },
+    [address, openConnectModal, currentChainId, config],
+  );
+
+  const { writeContractAsync, isPending: pendingFrameChange } = useWriteContract();
+  const { basenameChain } = useBasenameChain(profileUsername);
+  const setFrameRecord = useCallback(
+    async (frameUrl: string) => {
+      async function doTransaction() {
```

Comment:
If not using `euseWriteContractWithReceipt `, we should have error / analytics logging here